testing/unittests/jlibtests.cpp

```diff
@@ -3396,7 +3396,308 @@ element1: !el scalarvalue
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+        "<Configuration version=\"1.0\" environment=\"test\">"
+        "  <Database host=\"localhost\" port=\"3306\" ssl=\"true\">"
+        "    <Connection timeout=\"30\" pool=\"10\" retry=\"3\"/>"
+        "    <Tables>"
+        "      <Table name=\"users\" schema=\"public\" rows=\"1000\"/>"
+        "      <Table name=\"orders\" schema=\"sales\" rows=\"5000\"/>"
+        "    </Tables>"
+        "  </Database>"
+        "  <Cache enabled=\"true\" size=\"1024\" ttl=\"3600\">"
+        "    <Policies>"
+        "      <Policy name=\"default\" expiry=\"300\"/>"
+        "    </Policies>"
+        "  </Cache>"
+        "</Configuration>"};
+    Owned<IPropertyTree> originalTree;
+
+public:
+    void testSerializeVsSerializeToStreamRootOnlyPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createPTree("EmptyRoot"));
+    }
+
+    void testSerializeVsSerializeToStreamFromXmlPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createPTree(testXml));
+    }
+
+    void testSerializeVsSerializeToStreamFromYamlPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createYamlPropertyTree());
+    }
+
+    void testSerializeVsSerializeToStreamCPTValuePTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createCPTValuePTree());
+    }
+
+protected:
+    void testSerializeVsSerializeToStream(const char *testName, IPropertyTree *_originalTree)
+    {
+        originalTree.setown(_originalTree);
+
+        DBGLOG("Starting timing test for %s", testName);
+
+        // Time serialize() method
+        MemoryBuffer memBufOriginalTree;
+        CCycleTimer timer;
+        originalTree->serialize(memBufOriginalTree);
+        __uint64 serializedElapsedNs = timer.elapsedNs();
+
+        // Time serializeToStream() method
+        MemoryBuffer memoryBufferStream;
+        Owned<IBufferedSerialOutputStream> out = createBufferedSerialOutputStream(memoryBufferStream);
+        timer.reset();
+        originalTree->serializeToStream(*out);
+        out->flush();
+        __uint64 serializeToStreamElapsedNs = timer.elapsedNs();
+
+        // Convert to milliseconds for display
+        double serializeTimeMs = serializedElapsedNs / 1e6;
+        double serializeToStreamTimeMs = serializeToStreamElapsedNs / 1e6;
+
+        DBGLOG("Timing results for %s:", testName);
+        DBGLOG("  serialize() time: %.6f ms", serializeTimeMs);
+        DBGLOG("  serializeToStream() time: %.6f ms", serializeToStreamTimeMs);
+        DBGLOG("  Performance ratio (serializeToStream/serialize): %.6f", serializeToStreamTimeMs / serializeTimeMs);
+
+        // Compare memoryBufferStream to memBufOriginalTree using memcmp
+        CPPUNIT_ASSERT_EQUAL(memBufOriginalTree.length(), memoryBufferStream.length());
+        CPPUNIT_ASSERT(memcmp(memBufOriginalTree.toByteArray(), memoryBufferStream.toByteArray(), memBufOriginalTree.length()) == 0);
```

Comment:
quibble, no need to change, feels like these should be 1st, check worked, before timing/logging code.